x/clerk/keeper/grpc_query.go

```diff
@@ -0,0 +1,42 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/maticnetwork/heimdall/helper"
+	"github.com/maticnetwork/heimdall/x/clerk/types"
+)
+
+// Querier is used as Keeper will have duplicate methods if used directly, and gRPC names take precedence over keeper
+type Querier struct {
+	Keeper
+	contractCaller helper.IContractCaller
+}
+
+// NewQueryServerImpl returns an implementation of the bank MsgServer interface
+// for the provided Keeper.
+func NewQueryServerImpl(keeper Keeper, contractCaller helper.IContractCaller) types.QueryServer {
+	return &Querier{Keeper: keeper, contractCaller: contractCaller}
+}
+
+var _ types.QueryServer = Querier{}
+
+func (k Querier) Record(c context.Context, req *types.QueryRecordParams) (*types.QueryRecordResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	// get state record by record id
+	record, err := k.GetEventRecord(ctx, req.RecordId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryRecordResponse{EventRecord: record}, nil
+}
```

Comment:
Please add a new line at end of the file. Use VSCode settings for that -- https://stackoverflow.com/a/44704969

Reply:
i am using sublime text

Reply:
will check its setting and make changes